biz.aQute.bndlib/src/aQute/bnd/build/DownloadBlocker.java

```diff
@@ -110,4 +110,8 @@ public File getFile() {
 	public String toString() {
 		return "DownloadBlocker [stage=" + stage + ", failure=" + failure + ", file=" + file + "]";
 	}
+
+	public String getPath() {
+		return file.getAbsolutePath();
+	}
```

Comment:
Well we need a fix.. somehow we need the path of the file-to-be! Otherwise
we have an "equals" problem.

On Aug 25, 2016 6:11 PM, "BJ Hargrave" <notifications@github.com> wrote:

> In biz.aQute.bndlib/src/aQute/bnd/build/DownloadBlocker.java
> <https://github.com/bndtools/bnd/pull/1653#discussion_r76334542>:
>
> > @@ -110,4 +110,8 @@ public File getFile() {
> >  	public String toString() {
> >  		return "DownloadBlocker [stage=" + stage + ", failure=" + failure + ", file=" + file + "]";
> >  	}
> > +
> > +	public String getPath() {
> > +		return file.getAbsolutePath();
> > +	}
>
> This cannot work. The whole point is that the blocker does not have the
> file until sometime in the future. You cannot access the file field until
> getReason has been called.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bndtools/bnd/pull/1653/files/dd8e021dba5ad9f40aeb4c2eb69b9eca9c8b86fb#r76334542>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAI9TBRjphCsVlLcJZz2zChH-KYMAwueks5qjhMJgaJpZM4JthKq>
> .
>